spec/models/trip_spec.rb

```diff
@@ -0,0 +1,35 @@
+require "rails_helper"
+
+describe Trip do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:total_girls) }
+    it { is_expected.to validate_presence_of(:total_boys) }
+    it { is_expected.to validate_presence_of(:total_teachers) }
+    it { is_expected.to validate_presence_of(:total_bus_drivers) }
+    it { is_expected.to validate_presence_of(:start_date) }
+    it { is_expected.to validate_presence_of(:end_date) }
+
+    it { is_expected.to belong_to(:group) }
+    it { is_expected.to delegate_method(:name).to(:group).with_prefix }
+  end
+
+  context "when start_date is in the past" do
+    it "is not valid" do
+      trip = build(:trip, start_date: 1.month.ago)
+
+      expect(trip).not_to be_valid
+    end
+  end
+
+  context "when start_date is after end_date" do
+    it "is not valid" do
+      error_message = I18n.t(
+        "activerecord.errors.models.trip.start_date_after_end_date"
+      )
+      trip = build(:trip, start_date: 1.month.from_now, end_date: 2.month.ago)
+
+      expect(trip).not_to be_valid
+      expect(trip.errors[:start_date]).to eq [error_message]
```

Comment:
We usually strive to one assertion per spec. What do you think about splitting this spec into two?

Reply:
Done in `d8eb94a`